arangod/Network/NetworkFeature.cpp

```diff
@@ -238,4 +264,59 @@ void NetworkFeature::trackForwardedRequest() {
   ++_forwardedRequests;
 }
 
+std::size_t NetworkFeature::requestsInFlight() const {
+  return _requestsInFlight.load();
+}
+
+bool NetworkFeature::isCongested() const {
+  return _requestsInFlight.load() >= (_maxInFlight * ::CongestionRatio);
+}
+
+bool NetworkFeature::isSaturated() const {
+  return _requestsInFlight.load() >= _maxInFlight;
+}
+
+void NetworkFeature::sendRequest(network::ConnectionPool& pool,
+                                 network::RequestOptions const& options,
+                                 std::string const& endpoint,
+                                 std::unique_ptr<fuerte::Request>&& req,
+                                 RequestCallback&& cb) {
+  prepareRequest(pool, req);
+  auto conn = pool.leaseConnection(endpoint);
+  conn->sendRequest(std::move(req),
+                    [this, &pool,
+                     cb = std::move(cb)](fuerte::Error err,
+                                         std::unique_ptr<fuerte::Request> req,
+                                         std::unique_ptr<fuerte::Response> res) {
+                      finishRequest(pool, err, req, res);
+                      cb(err, std::move(req), std::move(res));
+                    });
+}
+
+void NetworkFeature::prepareRequest(network::ConnectionPool const& pool,
+                                    std::unique_ptr<fuerte::Request>& req) {
+  _requestsInFlight += 1;
```

Comment:
Can we relax the memory ordering here, e.g.
```suggestion
  _requestsInFlight.fetch_add(1, std::memory_order_relaxed);
```